cve_bin_tool/sbom_manager/parse.py

```diff
@@ -253,6 +254,12 @@ def parse_cyclonedx_spdx(self) -> [(str, str, str)]:
         sbom_parser = SBOMParser(sbom_type=self.type)
         # Load SBOM
         sbom_parser.parse_file(self.filename)
+        doc = sbom_parser.get_document()
+        uuid = doc.get("uuid", "")
+        parts = uuid.split(":")
+        if len(parts) == 3 and parts[0] == "urn" and parts[1] == "uuid":
+            self.serialNumber = parts[2]
```

Comment:
Is there any basic validation of the serial number that should be happening here?  Is there a list of allowed characters we should be checking?

Reply:
@terriko the serial number will include a-z and 0-9 with '-' in between, we can surely do validation the cyclonedx docs have a regex pattern for that but I am a little unsure what should we do if the serialNumber is not in the given format?

Reply:
Print out a message saying the serial number was invalid and halt, I think. If the user's got a really non-conformant SBOM they're going to have a bad time anyhow, and if it's just a little thing the user can fix it.

Reply:
And I'd be willing to change my stance on this if it turns out non-conformant serial numbers are rampant and we need to be more liberal in what we accept, but they're hopefully not?